skimage/future/graph/rag.py

```diff
@@ -461,20 +457,32 @@ def draw_rag(labels, rag, img, border_color=None, node_color='#ffff00',
     >>> img = data.coffee()
     >>> labels = segmentation.slic(img)
     >>> g =  graph.rag_mean_color(img, labels)
-    >>> out = graph.draw_rag(labels, g, img)
+    >>> smap = graph.show_rag(labels, g, img)
+    >>> cbar = plt.colorbar(smap)
     """
+
     if not in_place:
         rag = rag.copy()
 
-    if desaturate:
-        img = color.rgb2gray(img)
-        img = color.gray2rgb(img)
-
+    if ax is None:
+        fix, ax = plt.subplots()
     out = util.img_as_float(img, force_copy=True)
-    cc = colors.ColorConverter()
 
-    edge_color = cc.to_rgb(edge_color)
-    node_color = cc.to_rgb(node_color)
+    if img_cmap is None:
+        if img.ndim < 3:
+            raise ValueError('The image needs to be RGB')
+        if img.ndim < 4:
```

Comment:
What's all this? It seems brittle. If I give you a 3D grayscale image, you'll give me the first three columns blended as RGB and ignore the rest.

On the other hand, while we're assuming things, why not assume grayscale when the image is 2D and they've used colormap=None? Or, at least, perhaps the ValueError should specify that it should be RGB *or* a colormap should be selected.